code/datums/components/kneecapping.dm

```diff
@@ -0,0 +1,90 @@
+ */
+/datum/element/kneecapping
+	element_flags = ELEMENT_DETACH
+
+/datum/element/kneecapping/Attach(datum/target)
+	if(!isitem(target))
+		stack_trace("Kneecapping element added to non-item object: \[[target]\]")
+		return ELEMENT_INCOMPATIBLE
+
+	var/obj/item/target_item = target
+
+	if(target_item.force < WOUND_MINIMUM_DAMAGE)
+		stack_trace("Kneecapping element added to item with too little force to wound: \[[target]\]")
+		return ELEMENT_INCOMPATIBLE
+
+	. = ..()
+
+	if(. == ELEMENT_INCOMPATIBLE)
+		return
+
+	RegisterSignal(target, COMSIG_ITEM_ATTACK_SECONDARY , .proc/try_kneecap_target)
+
+/datum/element/kneecapping/Detach(datum/target)
+	UnregisterSignal(target, COMSIG_ITEM_ATTACK_SECONDARY)
+
+	return ..()
+
+/**
+ * Signal handler for COMSIG_ITEM_ATTACK_SECONDARY. Does checks for pacifism, zones and target state before either returning nothing
+ * if the special attack could not be attempted, performing the ordinary attack procs instead - Or cancelling the attack chain if
+ * the attack can be started.
+ */
+/datum/element/kneecapping/proc/try_kneecap_target(obj/item/source, mob/living/carbon/target, mob/attacker, params)
+	SIGNAL_HANDLER
+
+	if((attacker.zone_selected != BODY_ZONE_L_LEG) && (attacker.zone_selected != BODY_ZONE_R_LEG))
+		return
+
+	if(HAS_TRAIT(attacker, TRAIT_PACIFISM))
+		return
+
+	if(!iscarbon(target))
+		return
+
+	if(!target.buckled && !HAS_TRAIT(target, TRAIT_FLOORED) && !HAS_TRAIT(target, TRAIT_IMMOBILIZED))
+		return
+
+	var/obj/item/bodypart/leg = target.get_bodypart(attacker.zone_selected)
+
+	if(!leg)
+		return
+
+	. = COMPONENT_SECONDARY_CANCEL_ATTACK_CHAIN
+
+	INVOKE_ASYNC(src, .proc/do_kneecap_target, source, leg, target, attacker)
+
+/**
+ * After a short do_mob, attacker applies damage to the given leg with a significant wounding bonus, applying the weapon's force as damage.
+ */
+/datum/element/kneecapping/proc/do_kneecap_target(obj/item/weapon, obj/item/bodypart/leg, mob/living/carbon/target, mob/attacker)
+	if(LAZYACCESS(attacker.do_afters, weapon))
+		return
+
+	attacker.visible_message(span_warning("[attacker] carefully aims [attacker.p_their()] [weapon] for a swing at [target]'s kneecaps!"), span_danger("You carefully aim \the [weapon] for a swing at [target]'s kneecaps!"))
+	log_combat(attacker, target, "started aiming a swing to break the kneecaps of", weapon)
+
+	if(do_mob(attacker, target, 3 SECONDS, interaction_key = weapon))
+		attacker.visible_message(span_warning("[attacker] swings [attacker.p_their()] [weapon] at [target]'s kneecaps!"), span_danger("You swing \the [weapon] at [target]'s kneecaps!"))
+		var/datum/wound/blunt/severe/severe_wound_type = /datum/wound/blunt/severe
+		var/datum/wound/blunt/critical/critical_wound_type = /datum/wound/blunt/critical
+		leg.receive_damage(brute = weapon.force, wound_bonus = rand(initial(severe_wound_type.threshold_minimum), initial(critical_wound_type.threshold_minimum) + 10))
+		log_combat(attacker, target, "broke ther kneecaps of", weapon)
```

Comment:
ther kneecaps

Reply:
them thar kneecaps

Reply:
I haff marburrs in my mouf. Ther is no way I wud maek such a mistake.